jenkins/shared-library/vars/nightlyBuildPipeline.groovy

```diff
@@ -0,0 +1,177 @@
+	}
+}
+
+void triggerAllJobs(
+		String branchName,
+		boolean waitForDownStream,
+		String shouldPublishFailJobStatusName,
+		boolean shouldPublishFailJobStatusValue,
+		String manualGitBranchName) {
+	Map<String, String> siblingNameMap = siblingJobNames()
+	Map<String, String> displayNameJenkinsfileMap = jenkinsfileMap()
+	Map<String, Closure> buildJobs = [:]
+	final String platformName = 'PLATFORM'
+	String jobName = resolveJobName(siblingNameMap.keySet().toArray()[0], branchName)
+
+	siblingNameMap.each { siblingName ->
+		String displayName = siblingName.value
+		buildJobs["${displayName}"] = {
+			stage("${displayName}") {
+				String fullJobName = siblingName.key + '/' + jobName
+
+				// Platform parameter can vary per project, get the last value
+				echo "job name - ${fullJobName}"
+				String platform = jobParameterValueFromJenkinsfile(displayNameJenkinsfileMap.get(displayName), platformName.toLowerCase())
+				build job: "${fullJobName}", parameters: [
+					gitParameter(name: manualGitBranchName, value: branchName),
+					string(name: platformName, value: platform ?: 'ubuntu'),
+					string(name: 'BUILD_CONFIGURATION', value: 'release-private'),
+					string(name: 'TEST_MODE', value: 'code-coverage'),
+					booleanParam(name: 'SHOULD_PUBLISH_IMAGE', value: false),
+					booleanParam(name: shouldPublishFailJobStatusName, value: shouldPublishFailJobStatusValue)],
+					wait: waitForDownStream
+			}
+		}
+	}
+
+	parallel buildJobs
+}
+
+Map<String, String> siblingJobNames() {
+	Item project = Jenkins.get().getItemByFullName(currentBuild.fullProjectName)
+	List<Item> siblingItems = project.parent.items
+
+	Map<String, String> targets = [:]
+	for (Item item in siblingItems) {
+		if (org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject != item.getClass() || item.fullName == project.fullName) {
+			continue
+		}
+
+		targets.put(item.fullName, item.displayName)
+	}
+
+	return targets
+}
+
+String jobParameterValueFromJenkinsfile(String jenkinsfilePath, String parameterName) {
+	String value = sh(
+		script: "sed -rn \'s/${parameterName} = (.*)\$/\\1/p\' ${jenkinsfilePath}/Jenkinsfile",
+		returnStdout: true
+	)
+	value = value.replaceAll("(\\[|\\]|')", '').trim()
+	echo "Found ${parameterName} = ${value}"
+	return value
+}
+
+Map <String, String> jenkinsfileMap() {
+	Object buildConfiguration = yamlHelper.readYamlFromFile(helper.resolveBuildConfigurationFile())
+	Map<String, String> displayNameJenkinsfileMap = [:]
+
+	buildConfiguration.builds.each { build -> displayNameJenkinsfileMap.put(build.name, build.path) }
+	return displayNameJenkinsfileMap
+}
+
+String resolveJobName(String jobFolder, String branchName) {
+	final String seperator = '/'
+	String encodeJobName = branchName.replaceAll(seperator, '%2F')
+	String fullJobName = jobFolder + seperator + encodeJobName
+	Item job = Jenkins.get().getItemByFullName(fullJobName)
+	if (job?.isBuildable()) {
+		return encodeJobName
+	}
+
+	// find the PR that matches the branch
```

Comment:
why do we need this logic?

Reply:
After a pull requested is created, the Jenkins job name changes from the git branch name to the PR number.
E.g., this PR the Jenkins job name is ``Symbol/generated/symbol/jenkins/PR-373`` which matches the PR number above instead of the branch name `` feat/nightly_builds``.